packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -0,0 +1,211 @@
+   * To override the default value, provide height for the row that needs to be overridden.
+   */
+  rowHeight: PropTypes.string,
+  /**
+   * Number indicating the index of the column that represents row header. Index is 0 based and cannot exceed one less than the number of columns in the grid.
+   */
+  rowHeaderIndex: PropTypes.number,
+};
+
+const defaultProps = {
+  rowHeaderIndex: 0,
+};
+
+function WorklistDataGrid(props) {
+  const {
+    id,
+    ariaLabelledby,
+    ariaLabel,
+    columns,
+    rows,
+  } = props;
+
+  const focusedRow = useRef(0);
+  const focusedCol = useRef(0);
+  const grid = useRef();
+
+  useEffect(() => {
+    grid.current = document.getElementById(`${props.id}`);
+
+    const focusedCell = grid.current.rows[focusedRow.current].cells[focusedCol.current];
+    focusedCell.tabIndex = 0;
+  }, [props.id]);
+
+  const handleKeyDown = (event) => {
+    const currentFocusedRow = event.target.parentElement.rowIndex;
+    const currentFocusedCol = event.target.cellIndex;
+
+    let nextRow = currentFocusedRow;
+    let nextCol = currentFocusedCol;
+
+    const key = event.keyCode;
+    switch (key) {
+      case KeyCode.KEY_UP:
+        nextRow -= 1;
+        break;
+      case KeyCode.KEY_DOWN:
+        nextRow += 1;
+        break;
+      case KeyCode.KEY_LEFT:
+        nextCol -= 1;
+        break;
+      case KeyCode.KEY_RIGHT:
+        nextCol += 1;
+        break;
+      case KeyCode.KEY_HOME:
+        nextCol = 0;
+        if (event.ctrlKey) {
+          nextRow = 1; // Assumption is that the first row is the column Heading.
+        }
+        break;
+      case KeyCode.KEY_END:
+        nextCol = props.rows[0].cells.length - 1;
+        if (event.ctrlKey) {
+          nextRow = props.rows.length;
+        }
+        break;
+      default:
+        return;
+    }
+    if (nextRow >= grid.current.rows.length || nextCol >= grid.current.rows[0].cells.length) {
+      event.preventDefault();
+      return;
+    }
+    if (nextCol < 0 || nextRow < 0) {
+      event.preventDefault();
+      return;
+    }
+
+    // Remove the ability to receive visible focus as only one cell should be able to receive tab focus.
+    // This also ensures that tabbing back into the grid focuses on the cell that last had focus.
+    grid.current.rows[currentFocusedRow].cells[currentFocusedCol].tabIndex = -1;
+
+    focusedRow.current = nextRow;
+    focusedCol.current = nextCol;
+
+    const elementWithTabStop = grid.current.rows[nextRow].cells[nextCol];
+    elementWithTabStop.tabIndex = 0;
+    elementWithTabStop.focus();
+    event.preventDefault();
+  };
+
+  const handleClick = (event) => {
+    // Remove Tab stop from previous cell in table that has focus and set it to the cell that was clicked.
+    grid.current.rows[focusedRow.current].cells[focusedCol.current].tabIndex = -1;
+    focusedRow.current = event.target.parentElement.rowIndex;
+    focusedCol.current = event.target.cellIndex;
```

Comment:
Since you are using the target.  You will need to use something similar to the JavaScript closest API to get the proper td and tr elements.  This will make you logic future proof as well.  The closest method is not support in IE 11.  However, there are polyfills that can be used or you can create your own implementation.  You want to find the closest td and get the cellIndex.  Then use closest on the td element to find the parent tr element to get the rowIndex.  This logic would work in any scenario.

https://stackoverflow.com/questions/45656949/how-to-return-the-row-and-column-index-of-a-table-cell-by-clicking